### PR TITLE
fixed discovery service and security-doc errors

### PR DESCRIPTION
**Pull request by Spec163**

### CODE REVIEW

**Comment by kostua16** [comment]
Depends on #191 due to build failure for commit 6fe20a490eab0d4aaaf2a9ad5075c5e1fdafad1a

**Comment by kostua16** [comment]
Depends on #195 due to build failure for commit 6fe20a490eab0d4aaaf2a9ad5075c5e1fdafad1a

**Comment by Spec163** [comment]
я сломал тесты

**Comment by kostua16** [comment]
Depends on #200 due to build failure for commit b65ca5395db26caa5281782cb29d94b0768ae394

**Comment by kostua16** [comment]
Depends on #207 due to build failure for commit 8402a60460ffdd31d3b0237be0cb54af996d0079

**Comment by kostua16** [comment]
Depends on #209 due to build failure for commit bce90f75293d1ed80dcabcf34947025f386b040a

**Comment by kostua16** [comment]
Depends on #211 due to build failure for commit bce90f75293d1ed80dcabcf34947025f386b040a

**Comment by kostua16** [comment]
Depends on #212 due to build failure for commit bce90f75293d1ed80dcabcf34947025f386b040a

**Comment by kostua16** [comment]
Depends on #213 due to build failure for commit e2c3650e3cb0395d227fafd7204813f9c5368c2b

**Comment by kostua16** [comment]
Depends on #215 due to build failure for commit 9b0376a8842d2aad7299096d061aefae43f2feae

**Comment by kostua16** [comment]
Need to fix build for "client-ui" module by creating empty file "src/main/java/.gitkeep".

Error:  Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.8.0.2131:sonar (default) on project client-ui: The directory '/home/runner/work/UNC_2020_MS_TLT/UNC_2020_MS_TLT/client-ui/src/main/java' does not exist for Maven module nc.unc.cs:client-ui:jar:0.0.1-SNAPSHOT.

**Comment by kostua16** [comment]
Build successfully completed for commit d10379a853055a77949d60d9ecbb0e8394b5f681

File: services/tax/src/main/java/nc/unc/cs/services/tax/confogurations/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/tax/src/main/java/nc/unc/cs/services/tax/confogurations/SecurityConfig.java:26]: move to common module as it duplicated in each affected service

File: services/passport/src/main/java/nc/unc/cs/services/passport/confogurations/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/passport/src/main/java/nc/unc/cs/services/passport/confogurations/SecurityConfig.java:26]: to common module

File: services/logging/src/main/java/nc/unc/cs/services/logging/confogurations/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/logging/src/main/java/nc/unc/cs/services/logging/confogurations/SecurityConfig.java:26]: to common module

File: services/gibdd/src/main/java/nc/unc/cs/services/gibdd/confogurations/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/gibdd/src/main/java/nc/unc/cs/services/gibdd/confogurations/SecurityConfig.java:26]: to common module

File: services/communal/src/main/java/nc/unc/cs/services/communal/confogurations/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/communal/src/main/java/nc/unc/cs/services/communal/confogurations/SecurityConfig.java:26]: to common module

File: services/bank/src/main/java/nc/unc/cs/services/bank/confogurations/SecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        .and()
+        .authorizeRequests()
+        .antMatchers("/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html")
+        .permitAll()
+        .and()
+        .httpBasic();
+
+    super.configure(http);
+  }
+}
```
Comment (kostua16) [inline services/bank/src/main/java/nc/unc/cs/services/bank/confogurations/SecurityConfig.java:26]: to common module

File: pom.xml [hunk 7]
```diff
@@ -328,11 +348,11 @@
           <reportFormat>plain</reportFormat>
           <forkCount>1C</forkCount>
           <includes>
-            <include>**/*Test*.java</include>
+            <include>**services/*/services/*/**/*Test*.java</include>
             <include>**/*IT*.java</include>
           </includes>
           <!--suppress UnresolvedMavenProperty -->
-          <argLine>@{spring.runner.args} @{argLine}</argLine>
+          <argLine>${spring.runner.args} ${argLine}</argLine>
```
Comment (kostua16) [inline pom.xml:355]: > @{argLine}

is a special vadiable [@{argLine} meaning](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#argLine)

Comment (kostua16) [reply pom.xml:355]: >${spring.runner.args} - is correct change

**Comment by kostua16** [comment]
Depends on #227 due to build failure for commit 53ff96be50ffed792973ed353eade0dc92b4d618

**Comment by kostua16** [comment]
Build successfully completed for commit 788da2e084944c3a69b17b9ee3b786fb0bc5a78f

File: pom.xml (hunk 7 above)
Comment (kostua16) [reply pom.xml:355]: still not fixed

File: services/common/src/main/java/nc/unc/cs/services/common/configurations/SecurityConfig.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package nc.unc.cs.services.common.configurations;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
```
Comment (kostua16) [inline services/common/src/main/java/nc/unc/cs/services/common/configurations/SecurityConfig.java:10]: is it removed from other services?

**Review by kostua16 — CHANGES_REQUESTED** [review]

File: client-ui/src/services/communal/communal-service.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+                state: registration.state,
+                city: registration.city,
+                street: registration.street,
+                house: registration.house,
+                apartment: registration.apartment,
+                citizenId: registration.citizenId
+            }, {}
+        )
+            .then(response => {
+                    console.log('RESPONSE REGISTRATION: ', response)
```
Comment (kostua16) [inline client-ui/src/services/communal/communal-service.js:18]: todo: add UI feedback for the action call.

File: client-ui/src/services/communal/property-service.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+                city: creationProperty.city,
+                street: creationProperty.street,
+                house: creationProperty.house,
+                apartment: creationProperty.apartment,
+                apartmentSize: creationProperty.apartmentSize,
+                citizenId: creationProperty.citizenId
+            }, {}
+        )
+            .then(response => {
+                    console.log('REGISTRATE PROPERTY: ', response)
```
Comment (kostua16) [inline client-ui/src/services/communal/property-service.js:19]: todo: add UI feedback

File: client-ui/src/services/tax/tax-service.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import axios from 'axios'
+
+const API_URL = 'http://localhost:8082/tax'
```
Comment (kostua16) [inline client-ui/src/services/tax/tax-service.js:3]: todo: switch to node vue-dev server and will the property automatically from the configuration

**Review by kostua16 — CHANGES_REQUESTED** [review]

File: services/common/src/main/java/nc/unc/cs/services/common/configurations/SecurityConfig.java (hunk 8 above)
Comment (Spec163) [reply services/common/src/main/java/nc/unc/cs/services/common/configurations/SecurityConfig.java:10]: +

**Comment by kostua16** [comment]
Build successfully completed for commit ce0f250c18e8d7eca3ac4f6281c94937103bea37

**Review by kostua16 — APPROVED** [review]

**Pull request merged**